File: app/src/main/java/com/joshrap/liteweight/models/user/ReceivedWorkoutInfo.java
package com.joshrap.liteweight.models.user;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ReceivedWorkoutInfo {

    private String receivedWorkoutId;
    private String workoutName;
    private String receivedUtc;
    private boolean seen;
    private String senderId;
    private String senderUsername;
    private String senderProfilePicture;
    private Integer totalDays;
    private String mostFrequentFocus;
}

File: api/LiteWeightApi/Domain/Users/ReceivedWorkoutInfo.cs
using Google.Cloud.Firestore;
using LiteWeightAPI.Domain.Converters;
using NodaTime;

namespace LiteWeightAPI.Domain.Users;

[FirestoreData]
public class ReceivedWorkoutInfo
{
	[FirestoreProperty("receivedWorkoutId")]
	public required string ReceivedWorkoutId { get; set; }

	[FirestoreProperty("workoutName")]
	public required string WorkoutName { get; set; }

	[FirestoreProperty("receivedUtc", ConverterType = typeof(InstantConverter))]
	public Instant ReceivedUtc { get; set; }

	[FirestoreProperty("seen")]
	public bool Seen { get; set; }

	[FirestoreProperty("senderId")]
	public required string SenderId { get; set; }

	[FirestoreProperty("senderUsername")]
	public required string SenderUsername { get; set; }

	[FirestoreProperty("senderProfilePicture")]
	public required string SenderProfilePicture { get; set; }

	[FirestoreProperty("totalDays")]
	public int TotalDays { get; set; }

	[FirestoreProperty("mostFrequentFocus")]
	public required string MostFrequentFocus { get; set; }
}